cluster_agent/settings.py

```diff
@@ -63,6 +69,21 @@ def compute_extra_settings(cls, values):
         if values["SINGLE_USER_SUBMITTER"] is None:
             values["SINGLE_USER_SUBMITTER"] = values["X_SLURM_USER_NAME"]
 
+        assert any(
```

Comment:
Better to raise an explicit exception here instead of relying on assert.

Reply:
I was looking to the built-in exceptions and none of them make sense, I'd need to create another specifically for that. It's common in pydantic docs to use AssertionError here so that's why I decided for it

Reply:
I think it's best practice to not use assertions to check conditions in production code because assertions can be disabled depending on the execution mode of python.

You could use pydantic's `ValidationError` as the exception type. I think that would fit very well.

Reply:
I looked at the `ValidationError` class first and this class requires a pydantic model as parameter. Not sure if it is useful in this context. So, I went to [pydantic's docs](https://pydantic-docs.helpmanual.io/usage/validators/) and found examples with assertion. Indeed, not my favourite but I'd guess it is the recommended. Does it make sense for you?

Also, to disable the assertion error the python code must run with the `-O` flag, which isn't the case here because the agent is supposed to run with the `agentrun` entrypoint defined in the [setup.py](https://github.com/omnivector-solutions/cluster-agent/blob/main/setup.py#L55) file, and indeed, [is what happens in our charm](https://github.com/omnivector-solutions/charm-cluster-agent/blob/main/src/templates/ovs-cluster-agent.service#L10).

Reply:
I saw this comment I came to see because I'm using assert in one of my drafts. A way I see to go around this is using [check-expressions](https://py-buzz.readthedocs.io/en/latest/features.html#check-expressions), from py-buzz.

Reply:
it would be the same as `if condition: raise Error`, which is basically what the AssertionError here is doing in a different way